src/crewai/utilities/events/crewai_event_bus.py

```diff
@@ -68,14 +68,12 @@ def emit(self, source: Any, event: CrewEvent) -> None:
             event: The event instance to emit
         """
         event_type = type(event)
-        if event_type in self._handlers:
-            for handler in self._handlers[event_type]:
-                handler(source, event)
-        self._signal.send(source, event=event)
+        for event_type, handlers in self._handlers.items():
+            if isinstance(event, event_type):
+                for handler in handlers:
+                    handler(source, event)
 
-    def clear_handlers(self) -> None:
-        """Clear all registered event handlers - useful for testing"""
-        self._handlers.clear()
```

Comment:
The event bus is a singleton, meaning even if you instantiate a new object, it's reusing the existing one. For that reason, calling this method in tests would make other tests fail. It wasn't being used anywhere else, so I removed it.